app/scripts/migrations/138.ts

```diff
@@ -0,0 +1,430 @@
+          new Error(
+            `Migration ${version}: typeof state.NetworkController.networkConfigurationsByChainId["${chainId}"] is ${typeof networkConfiguration}`,
+          ),
+        );
+        malformedDataErrorFound = true;
+        continue;
+      }
+      if (!Array.isArray(networkConfiguration.rpcEndpoints)) {
+        global.sentry?.captureException(
+          new Error(
+            `Migration ${version}: typeof state.NetworkController.networkConfigurationsByChainId["${chainId}"].rpcEndpoints is ${typeof networkConfiguration.rpcEndpoints}`,
+          ),
+        );
+        malformedDataErrorFound = true;
+        continue;
+      }
+
+      for (const rpcEndpoint of networkConfiguration.rpcEndpoints) {
+        if (!isObject(rpcEndpoint)) {
+          global.sentry?.captureException(
+            new Error(
+              `Migration ${version}: typeof state.NetworkController.networkConfigurationsByChainId["${chainId}"].rpcEndpoints[] is ${typeof rpcEndpoint}`,
+            ),
+          );
+          malformedDataErrorFound = true;
+          continue;
+        }
+
+        if (rpcEndpoint.networkClientId === networkClientId) {
+          matchingChainId = chainId;
+        }
+      }
+    }
+    if (malformedDataErrorFound) {
+      return undefined;
+    }
+
+    if (matchingChainId) {
+      return matchingChainId;
+    }
+
+    const builtInChainId = BUILT_IN_NETWORKS.get(networkClientId);
+    if (!builtInChainId) {
+      global.sentry?.captureException(
+        new Error(
+          `Migration ${version}: No chainId found for ${propertyName} "${networkClientId}"`,
+        ),
+      );
+    }
+    return builtInChainId;
+  };
+
+  const currentChainId = getChainIdForNetworkClientId(
+    selectedNetworkClientId,
+    'selectedNetworkClientId',
+  );
+  if (!currentChainId) {
+    return oldState;
+  }
+
+  // perform mutations on the cloned state
+  for (const [origin, subject] of Object.entries(subjects)) {
+    if (!isObject(subject)) {
+      global.sentry?.captureException?.(
+        new Error(
+          `Migration ${version}: Invalid subject for origin "${origin}" of type ${typeof subject}`,
+        ),
+      );
+      return oldState;
+    }
+
+    if (
+      !hasProperty(subject, 'permissions') ||
+      !isObject(subject.permissions)
+    ) {
+      global.sentry?.captureException?.(
+        new Error(
+          `Migration ${version}: Invalid permissions for origin "${origin}" of type ${typeof subject.permissions}`,
+        ),
+      );
+      return oldState;
+    }
+
+    const { permissions } = subject;
+
+    let basePermission: PermissionConstraint | undefined;
+
+    let ethAccounts: string[] = [];
+    const ethAccountsPermission = permissions[PermissionNames.eth_accounts];
+    const permittedChainsPermission =
+      permissions[PermissionNames.permittedChains];
+
+    // if there is no eth_accounts permission we can't create a valid CAIP-25 permission so we remove the permission
```

Comment:
We should mention this in the PR description as an additional functional change

Reply:
Sounds good!

Reply:
Added the following disclaimer in the PR description: 

> - **CHANGED**: for dapps that were granted chain permissions via the `wallet_addEthereum` or `wallet_switchEthereumChain` flows without account permissions, these permissions will be removed with this migration. We think this ok because: 
>    - This is a very uncommon scenario for dapps to request chain switches without account permissions. 
>    - These permissions can be regained very trivially with subsequent chain switch requests.
>